### PR TITLE
gate: add price_type ccxt/ccxt#16749

### PR DESCRIPTION
**Pull request by sc0Vu**
Fix ccxt/ccxt#16749

Add `price_type` in params.

```BASH
$ node examples/js/cli gateio createOrder ADA/USDT:USDT limit buy 1 0.02 '{"price_type":2,"stopLossPrice":1}' --verbose
Node.js: v14.17.0
CCXT v2.7.76
gateio.createOrder (ADA/USDT:USDT, limit, buy, 1, 0.02, [object Object])
fetch Request:
 gateio POST https://api.gateio.ws/api/v4/futures/usdt/price_orders 
RequestHeaders:
 {
  'X-Gate-Channel-Id': 'ccxt',
  KEY: 'xxxxxx',
  SIGN: 'xxxxxx',
  'Content-Type': 'application/json'
} 
RequestBody:
 {"initial":{"contract":"ADA_USDT","size":1,"price":"0.02"},"trigger":{"price_type":2,"price":"1","rule":1}} 

handleRestResponse:
 gateio POST https://api.gateio.ws/api/v4/futures/usdt/price_orders 201 Created 
ResponseHeaders:
 {
  'Access-Control-Allow-Headers': 'DNT,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type',
  'Access-Control-Allow-Methods': 'GET,POST,OPTIONS,DELETE,PUT',
  'Access-Control-Max-Age': '1728000',
  Connection: 'keep-alive',
  'Content-Length': '15',
  'Content-Type': 'application/json',
  Server: 'openresty',
  Vary: 'Origin'
} 
ResponseBody:
 {"id":xxxxxx} 

2023-02-10T16:01:58.468Z iteration 0 passed in 70 ms

{
  id: 'xxxxxx',
  clientOrderId: undefined,
  timestamp: undefined,
  datetime: undefined,
  lastTradeTimestamp: undefined,
  status: undefined,
  symbol: 'ADA/USDT:USDT',
  type: undefined,
  timeInForce: undefined,
  postOnly: false,
  reduceOnly: undefined,
  side: undefined,
  price: undefined,
  stopPrice: undefined,
  triggerPrice: undefined,
  average: undefined,
  amount: undefined,
  cost: undefined,
  filled: undefined,
  remaining: undefined,
  fee: undefined,
  fees: [],
  trades: [],
  info: { id: 'xxxxxx' }
}
```

There are some other exchanges support different price type (last price / mark price / index price) in stop orders. Shall we consider to unify this in parameters?

### CODE REVIEW

**Comment by carlosmiei** [comment]
[🤖 TestsManager 🤖  Id: 63e676ddff417f917201ae43]
 ```

 Keys not found for gate, skipping tests!
```

File: js/gate.js [hunk 1]
```diff
@@ -3264,9 +3264,14 @@ module.exports = class gate extends Exchange {
                         rule = (side === 'buy') ? 2 : 1;
                         triggerOrderPrice = this.priceToPrecision (symbol, takeProfitPrice);
                     }
+                    const priceType = this.safeInteger (params, 'price_type', 0);
```
Comment (pcriadoperez) [inline js/gate.js:3267]: LGTM, though I'd add the param to the docstring of the function

**Comment by carlosmiei** [comment]
LGTM ✅ 

```
 p gate createOrder ADA/USDT:USDT limit buy 1 0.02 '{"stopLossPrice":1}'
Python v3.10.9
CCXT v2.7.76
gate.createOrder(ADA/USDT:USDT,limit,buy,1,0.02,{'stopLossPrice': 1})
{'amount': None,
 'average': None,
 'clientOrderId': None,
 'cost': None,
 'datetime': None,
 'fee': None,
 'fees': [],
 'filled': None,
 'id': '36828125',
 'info': {'id': '36828125'},
 'lastTradeTimestamp': None,
 'postOnly': False,
 'price': None,
 'reduceOnly': None,
 'remaining': None,
 'side': None,
 'status': None,
 'stopPrice': None,
 'symbol': 'ADA/USDT:USDT',
 'timeInForce': None,
 'timestamp': None,
 'trades': [],
 'triggerPrice': None,
 'type': None}
```

**Review by carlosmiei — APPROVED** [review]

**Comment by carlosmiei** [comment]
[🤖 TestsManager 🤖  Id: 63e777ebff417f917201ae52]
 ```

 Keys not found for gate, skipping tests!
```

**Pull request merged**